shared/utility/math/filter/ParticleFilter.hpp

```diff
@@ -26,8 +26,42 @@
 #include <random>
 #include <vector>
 
+#include "utility/math/stats/multivariate.hpp"
+#include "utility/math/stats/resample/multinomial.hpp"
+#include "utility/math/stats/resample/resample.hpp"
+#include "utility/math/stats/resample/residual.hpp"
+#include "utility/math/stats/resample/stratified.hpp"
+#include "utility/math/stats/resample/systematic.hpp"
+
 namespace utility::math::filter {
 
+    using utility::math::stats::MultivariateNormal;
+
+    // The possible resampling methods
+    // See http://users.isy.liu.se/rt/schon/Publications/HolSG2006.pdf for an explanation on each one
+    enum class ResampleMethod { MULTINOMIAL, RESIDUAL, STRATIFIED, SYSTEMATIC };
+
+    /**
+     * @author Alex Biddulph, Trent Houliston
+     *
+     * @details Uses a set of particles (or samples) to represent possible states in a system. Each particle is randomly
+     * selected from a multivariate normal distribution. Each particle is weighted by how likely it is to be the real
+     * state of the system. Particles with a higher weight are more likely to be resampled (with noise) during the time
+     * update.
+     *
+     * Time Update: Resamples all particles, perturbing the resampled particles with the model's process noise, and asks
+     * the model to predict what should happen to each particle after an elapsed amount of time. Also known as the
+     * predict step.
+     *
+     * Measurement Update: Asks the model to predict what the next measurement will be (based on current state/particle)
+     * and the finds the difference between the model's prediction and the actual measurement. The likelihood of this
+     * measurement occuring is then determined using a multivariate normal distribution probability density function.
```

Comment:
It's actually the other way around, it's calculating the likelihood of this model's prediction given the measurement